### PR TITLE
Limit pay button

### PR DESCRIPTION
**Pull request by wraeth-eth**
## What does this PR do and why?

Closes https://github.com/jbx-protocol/juice-interface/issues/1001

Also updated antd to get new functionality for inputs.

## Screenshots or screen recordings

https://user-images.githubusercontent.com/104132113/168269649-3453c5d7-9fa2-4ae3-824a-4d0f1b0871d0.mov

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "@uniswap/sdk-core": "3.0.1",
     "@uniswap/v3-sdk": "3.6.3",
     "@walletconnect/web3-provider": "^1.5.4",
-    "antd": "4.16.13",
+    "antd": "4.18.0",
```
Comment (johnnyd-eth) [inline package.json:25]: **Question**

**Review by johnnyd-eth — APPROVED** [review]
Clean AF. So good. Just wondering if the Antd update should be in this PR?

File: package.json (hunk 1 above)
Comment (wraeth-eth) [reply package.json:25]: Thanks for the review! The antd update brings in the change to the input form to use showCount. It was added in 4.7.x.

File: src/constants/numbers.ts [hunk 2]
```diff
@@ -2,3 +2,5 @@ export const WAD_DECIMALS = 18
 
 export const SECONDS_IN_DAY = 24 * 60 * 60
 export const SECONDS_IN_HOUR = 60 * 60
+
+export const PROJECT_PAY_CHARACTER_LIMIT = 16
```
Comment (aeolianeth) [inline src/constants/numbers.ts:6]: @wraeth-eth given https://github.com/jbx-protocol/juice-interface/pull/1007, would it make sense to increase this slightly? Not sure, your call 🙏

Comment (wraeth-eth) [reply src/constants/numbers.ts:6]: Oh for sure - if we have plans to wrap it, lets increase maybe to 32? WDYT?

Comment (wraeth-eth) [reply src/constants/numbers.ts:6]: Could even go bigger tbh:
image

Comment (wraeth-eth) [reply src/constants/numbers.ts:6]: image

Comment (aeolianeth) [reply src/constants/numbers.ts:6]: On second thought I think I'm in favour of keeping it at 16. Not sure we want to encourage long buttons, generally not smart from UX pov.

In other words, lgtm!

Comment (wraeth-eth) [reply src/constants/numbers.ts:6]: 🙌

**Review by aeolianeth — APPROVED** [review]
@wraeth-eth approving and merging!

@johnnyd-eth lets keep an eye out for any ant-d weirdness that might have been introduce from the upgrade.

**Pull request merged**